### PR TITLE
Allow rotating the progress indicator to indicate that a new track has started

### PR DESCRIPTION
**Pull request by schmiphi**
#### WHAT
Add a Flow<Unit> that triggers a rotation of the progress indicator.

#### WHY
This can be used to indicate to the user that a new media track has started.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt [hunk 1]
```diff
@@ -181,11 +187,19 @@ public fun AnimatedPlayPauseProgressButton(
...
     val animatedProgressColor = animateColorAsState(
-        targetValue = progressColor,
-        animationSpec = tween(450, 0, LinearEasing)
+        targetValue = progressColor, animationSpec = tween(450, 0, LinearEasing)
+    )
+
+    val progressIndicatorRotation by produceState(0f, rotateProgressIndicator) {
+        rotateProgressIndicator?.collectLatest { value += 360 }
+    }
+    val animatedProgressIndicatorRotation by animateFloatAsState(
```
Comment (yschimke) [inline media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt:200]: That's really nice. 

As a style nit, I might suggest pulling it out to a single function, animateChangeAsRotation? But your call.

**Comment by yschimke** [comment]
I'll leave @fstanis to review, but just curious.

File: media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt [hunk 2]
```diff
@@ -181,11 +187,19 @@ public fun AnimatedPlayPauseProgressButton(
...
+        targetValue = progressIndicatorRotation,
+        animationSpec = ProgressIndicatorDefaults.ProgressAnimationSpec
```
[the remark this answers is not in the document]
Comment (schmiphi) [reply media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt:203]: The spec needs to be the same as the spec for animating the progress itself, as the two animations will happen in parallel. This animation is actually a bit too slow (not stiff enough), so I'll send a separate PR to change the progress animation spec and the rotation animation spec together.

File: media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt (hunk 1 above)
Comment (schmiphi) [reply media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt]: Sounds good, done.

File: media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt [hunk 3]
```diff
@@ -218,3 +228,15 @@ public fun AnimatedPlayPauseProgressButton(
...
 }
+
+@Composable
+private fun animateChangeAsRotation(rotateProgressIndicator: Flow<Unit>): Float {
+    val progressIndicatorRotation by produceState(0f, rotateProgressIndicator) {
+        rotateProgressIndicator.collectLatest { value += 360 }
+    }
+    val animatedProgressIndicatorRotation by animateFloatAsState(
+        targetValue = progressIndicatorRotation,
+        animationSpec = ProgressIndicatorDefaults.ProgressAnimationSpec
```
Comment (yschimke) [inline media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt:239]: Consider settings a label, so it shows up in Android Studio tooling.

Comment (schmiphi) [reply media/ui/src/main/java/com/google/android/horologist/media/ui/components/animated/AnimatedPlayPauseButton.kt]: Done

**Review by fstanis — APPROVED** [review]

**Pull request merged**